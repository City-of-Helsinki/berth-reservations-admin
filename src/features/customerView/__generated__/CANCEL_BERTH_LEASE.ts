/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { TerminateBerthLeaseMutationInput } from "./../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL mutation operation: CANCEL_BERTH_LEASE
// ====================================================

export interface CANCEL_BERTH_LEASE_terminateBerthLease {
  __typename: "TerminateBerthLeaseMutationPayload";
  clientMutationId: string | null;
}

export interface CANCEL_BERTH_LEASE {
  terminateBerthLease: CANCEL_BERTH_LEASE_terminateBerthLease | null;
}

export interface CANCEL_BERTH_LEASEVariables {
  input: TerminateBerthLeaseMutationInput;
}
