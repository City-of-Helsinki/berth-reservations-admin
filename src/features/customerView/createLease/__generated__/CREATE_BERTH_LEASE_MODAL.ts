/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: CREATE_BERTH_LEASE_MODAL
// ====================================================

export interface CREATE_BERTH_LEASE_MODAL_profile_boats_edges_node {
  __typename: "BoatNode";
  id: string;
  name: string;
}

export interface CREATE_BERTH_LEASE_MODAL_profile_boats_edges {
  __typename: "BoatNodeEdge";
  node: CREATE_BERTH_LEASE_MODAL_profile_boats_edges_node | null;
}

export interface CREATE_BERTH_LEASE_MODAL_profile_boats {
  __typename: "BoatNodeConnection";
  edges: (CREATE_BERTH_LEASE_MODAL_profile_boats_edges | null)[];
}

export interface CREATE_BERTH_LEASE_MODAL_profile {
  __typename: "ProfileNode";
  boats: CREATE_BERTH_LEASE_MODAL_profile_boats | null;
}

export interface CREATE_BERTH_LEASE_MODAL_harbors_edges_node_properties {
  __typename: "HarborProperties";
  name: string | null;
}

export interface CREATE_BERTH_LEASE_MODAL_harbors_edges_node {
  __typename: "HarborNode";
  id: string;
  properties: CREATE_BERTH_LEASE_MODAL_harbors_edges_node_properties | null;
}

export interface CREATE_BERTH_LEASE_MODAL_harbors_edges {
  __typename: "HarborNodeEdge";
  node: CREATE_BERTH_LEASE_MODAL_harbors_edges_node | null;
}

export interface CREATE_BERTH_LEASE_MODAL_harbors {
  __typename: "HarborNodeConnection";
  edges: (CREATE_BERTH_LEASE_MODAL_harbors_edges | null)[];
}

export interface CREATE_BERTH_LEASE_MODAL {
  profile: CREATE_BERTH_LEASE_MODAL_profile | null;
  harbors: CREATE_BERTH_LEASE_MODAL_harbors | null;
}

export interface CREATE_BERTH_LEASE_MODALVariables {
  customerId: string;
}
