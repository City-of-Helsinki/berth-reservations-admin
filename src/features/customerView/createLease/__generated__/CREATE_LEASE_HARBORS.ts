/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: CREATE_LEASE_HARBORS
// ====================================================

export interface CREATE_LEASE_HARBORS_harbors_edges_node_properties {
  __typename: "HarborProperties";
  name: string | null;
}

export interface CREATE_LEASE_HARBORS_harbors_edges_node {
  __typename: "HarborNode";
  id: string;
  properties: CREATE_LEASE_HARBORS_harbors_edges_node_properties | null;
}

export interface CREATE_LEASE_HARBORS_harbors_edges {
  __typename: "HarborNodeEdge";
  node: CREATE_LEASE_HARBORS_harbors_edges_node | null;
}

export interface CREATE_LEASE_HARBORS_harbors {
  __typename: "HarborNodeConnection";
  edges: (CREATE_LEASE_HARBORS_harbors_edges | null)[];
}

export interface CREATE_LEASE_HARBORS {
  harbors: CREATE_LEASE_HARBORS_harbors | null;
}
