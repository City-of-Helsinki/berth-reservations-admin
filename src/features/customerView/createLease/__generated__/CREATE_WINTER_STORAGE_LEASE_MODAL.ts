/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: CREATE_WINTER_STORAGE_LEASE_MODAL
// ====================================================

export interface CREATE_WINTER_STORAGE_LEASE_MODAL_profile_boats_edges_node {
  __typename: "BoatNode";
  id: string;
  name: string;
}

export interface CREATE_WINTER_STORAGE_LEASE_MODAL_profile_boats_edges {
  __typename: "BoatNodeEdge";
  node: CREATE_WINTER_STORAGE_LEASE_MODAL_profile_boats_edges_node | null;
}

export interface CREATE_WINTER_STORAGE_LEASE_MODAL_profile_boats {
  __typename: "BoatNodeConnection";
  edges: (CREATE_WINTER_STORAGE_LEASE_MODAL_profile_boats_edges | null)[];
}

export interface CREATE_WINTER_STORAGE_LEASE_MODAL_profile {
  __typename: "ProfileNode";
  /**
   * The ID of the object
   */
  id: string;
  boats: CREATE_WINTER_STORAGE_LEASE_MODAL_profile_boats | null;
}

export interface CREATE_WINTER_STORAGE_LEASE_MODAL_winterStorageAreas_edges_node_properties {
  __typename: "WinterStorageAreaProperties";
  name: string | null;
}

export interface CREATE_WINTER_STORAGE_LEASE_MODAL_winterStorageAreas_edges_node {
  __typename: "WinterStorageAreaNode";
  id: string;
  properties: CREATE_WINTER_STORAGE_LEASE_MODAL_winterStorageAreas_edges_node_properties | null;
}

export interface CREATE_WINTER_STORAGE_LEASE_MODAL_winterStorageAreas_edges {
  __typename: "WinterStorageAreaNodeEdge";
  node: CREATE_WINTER_STORAGE_LEASE_MODAL_winterStorageAreas_edges_node | null;
}

export interface CREATE_WINTER_STORAGE_LEASE_MODAL_winterStorageAreas {
  __typename: "WinterStorageAreaNodeConnection";
  edges: (CREATE_WINTER_STORAGE_LEASE_MODAL_winterStorageAreas_edges | null)[];
}

export interface CREATE_WINTER_STORAGE_LEASE_MODAL {
  /**
   * Get profile by profile ID.
   * 
   * Requires `staff` credentials for the requester's service.The profile must have an active connection to the requester's service, otherwise it will not be returned.
   */
  profile: CREATE_WINTER_STORAGE_LEASE_MODAL_profile | null;
  winterStorageAreas: CREATE_WINTER_STORAGE_LEASE_MODAL_winterStorageAreas | null;
}

export interface CREATE_WINTER_STORAGE_LEASE_MODALVariables {
  customerId: string;
}
