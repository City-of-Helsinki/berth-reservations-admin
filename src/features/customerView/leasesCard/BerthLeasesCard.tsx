import React from 'react';
import { useTranslation } from 'react-i18next';

import { BerthLease } from '../types';
import LeasesCard from './LeasesCard';

export interface BerthLeasesCardProps {
  leases: BerthLease[];
  handleShowContract(leaseId: string): void;
}

const BerthLeasesCard = ({ leases, handleShowContract }: BerthLeasesCardProps) => {
  const { t } = useTranslation();
  const leaseDetails = leases.map((lease) => {
    return {
      id: lease.id,
      endDate: lease.endDate,
      startDate: lease.startDate,
      link: lease.harbor ? `/harbors/${lease.harbor.id}` : undefined,
      address: `${lease.harbor?.name || ''} ${lease.pierIdentifier || ''} ${lease.berthNum}`,
    };
  });

  return (
    <LeasesCard
      leaseDetails={leaseDetails}
      title={t('customerView.leases.berth.title')}
      infoSectionTitle={t('customerView.leases.berth.infoSectionTitle')}
      addressLabel={t('customerView.leases.berth.addressLabel')}
      handleShowContract={handleShowContract}
    />
  );
};

export default BerthLeasesCard;
