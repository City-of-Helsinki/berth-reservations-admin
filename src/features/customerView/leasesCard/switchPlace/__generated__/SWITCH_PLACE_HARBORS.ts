/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: SWITCH_PLACE_HARBORS
// ====================================================

export interface SWITCH_PLACE_HARBORS_harbors_edges_node_properties {
  __typename: "HarborProperties";
  name: string | null;
}

export interface SWITCH_PLACE_HARBORS_harbors_edges_node {
  __typename: "HarborNode";
  id: string;
  properties: SWITCH_PLACE_HARBORS_harbors_edges_node_properties | null;
}

export interface SWITCH_PLACE_HARBORS_harbors_edges {
  __typename: "HarborNodeEdge";
  node: SWITCH_PLACE_HARBORS_harbors_edges_node | null;
}

export interface SWITCH_PLACE_HARBORS_harbors {
  __typename: "HarborNodeConnection";
  edges: (SWITCH_PLACE_HARBORS_harbors_edges | null)[];
}

export interface SWITCH_PLACE_HARBORS {
  harbors: SWITCH_PLACE_HARBORS_harbors | null;
}
