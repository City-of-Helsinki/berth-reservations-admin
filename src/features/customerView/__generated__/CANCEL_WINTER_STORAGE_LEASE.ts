/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { TerminateWinterStorageLeaseMutationInput } from "./../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL mutation operation: CANCEL_WINTER_STORAGE_LEASE
// ====================================================

export interface CANCEL_WINTER_STORAGE_LEASE_terminateWinterStorageLease {
  __typename: "TerminateWinterStorageLeaseMutationPayload";
  clientMutationId: string | null;
}

export interface CANCEL_WINTER_STORAGE_LEASE {
  terminateWinterStorageLease: CANCEL_WINTER_STORAGE_LEASE_terminateWinterStorageLease | null;
}

export interface CANCEL_WINTER_STORAGE_LEASEVariables {
  input: TerminateWinterStorageLeaseMutationInput;
}
