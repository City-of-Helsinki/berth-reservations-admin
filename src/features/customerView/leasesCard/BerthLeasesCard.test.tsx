import React from 'react';
import { mount } from 'enzyme';
import { HashRouter } from 'react-router-dom';

import BerthLeasesCard, { BerthLeasesCardProps } from './BerthLeasesCard';
import { BerthMooringType, LeaseStatus } from '../../../@types/__generated__/globalTypes';

jest.mock('../../contractDetails/BerthContractDetailsContainer', () => {
  const BerthContractDetailsContainer = () => <div>Mocked</div>;

  return {
    __esModule: true,
    default: BerthContractDetailsContainer,
  };
});

describe('BerthLeasesCard', () => {
  const initialProps: BerthLeasesCardProps = {
    cancelLease: jest.fn(),
    createLease: jest.fn(),
    customerName: 'Tester McTestington',
    leases: [
      {
        berthNum: '1',
        depth: 6,
        endDate: '2019-09-10',
        harbor: {
          id: 'MOCK-HARBOR-1',
          name: 'Pursilahdenranta',
        },
        id: '1234',
        isActive: true,
        length: 6,
        mooringType: BerthMooringType.SINGLE_SLIP_PLACE,
        pierIdentifier: '7G',
        startDate: '2019-06-14',
        status: LeaseStatus.DRAFTED,
        width: 3,
      },
      {
        berthNum: '1',
        depth: null,
        endDate: '2020-06-15',
        harbor: {
          id: 'MOCK-HARBOR-2',
          name: 'Telakkakatu 1',
        },
        id: '4321',
        isActive: false,
        length: 6,
        mooringType: BerthMooringType.SIDE_SLIP_PLACE,
        pierIdentifier: 'A',
        startDate: '2019-05-20',
        status: LeaseStatus.TERMINATED,
        width: 3,
      },
    ],
  };

  const getWrapper = (props?: Partial<BerthLeasesCardProps>) =>
    mount(
      <HashRouter>
        <BerthLeasesCard {...initialProps} {...props} />
      </HashRouter>
    );

  beforeEach(() => {
    jest.restoreAllMocks();
  });

  it('renders normally', () => {
    const wrapper = getWrapper();

    expect(wrapper.render()).toMatchSnapshot();
  });
});
