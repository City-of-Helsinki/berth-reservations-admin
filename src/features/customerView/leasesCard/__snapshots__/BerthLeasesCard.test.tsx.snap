// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BerthLeasesCard renders normally 1`] = `
<div
  class="card"
>
  <div
    class="cardHeader"
  >
    <span
      class="title"
    >
      VENEPAIKAT
    </span>
  </div>
  <div
    class="cardBody"
  >
    <span
      class="StatusLabel-module_statusLabel__3R2J2 status-label_hds-status-label__1L8YI status"
    >
      Luotu
    </span>
    <article
      class="section"
    >
      <h4
        class="text standard h4 m title"
      >
        VENEPAIKKA
      </h4>
      <section
        class="body"
      >
        <div
          class="labelValuePair"
        >
          <span
            class="label standard"
          >
            Satama ja paikka:
          </span>
          <span
            class="value left"
          >
            <a
              class="internalLink brand"
              href="#/harbors/MOCK-HARBOR-1"
              title="Pursilahdenranta 7G 1"
            >
              Pursilahdenranta 7G 1
            </a>
          </span>
        </div>
        <div
          class="labelValuePair"
        >
          <span
            class="label standard"
          >
            Pituus:
          </span>
          <span
            class="value left"
          >
            6,0 m
          </span>
        </div>
        <div
          class="labelValuePair"
        >
          <span
            class="label standard"
          >
            Leveys:
          </span>
          <span
            class="value left"
          >
            3,0 m
          </span>
        </div>
        <div
          class="labelValuePair"
        >
          <span
            class="label standard"
          >
            Syvyys:
          </span>
          <span
            class="value left"
          >
            6,0 m
          </span>
        </div>
        <div
          class="labelValuePair"
        >
          <span
            class="label standard"
          >
            Kiinnitystapa:
          </span>
          <span
            class="value left"
          >
            Aisapaikka
          </span>
        </div>
        <div
          class="labelValuePair"
        >
          <span
            class="label standard"
          >
            Voimassa:
          </span>
          <span
            class="value left"
          >
            14.06.2019 - 10.09.2019
          </span>
        </div>
      </section>
    </article>
    <div>
      Mocked
    </div>
  </div>
  <div
    class="cardBody"
  >
    <span
      class="StatusLabel-module_statusLabel__3R2J2 status-label_hds-status-label__1L8YI StatusLabel-module_error__zNgn8 status-label_hds-status-label--error__2EYi4 status"
    >
      Irtisanottu
    </span>
    <article
      class="section"
    >
      <h4
        class="text standard h4 m title"
      >
        VENEPAIKKA
      </h4>
      <section
        class="body"
      >
        <div
          class="labelValuePair"
        >
          <span
            class="label standard"
          >
            Satama ja paikka:
          </span>
          <span
            class="value left"
          >
            <a
              class="internalLink brand"
              href="#/harbors/MOCK-HARBOR-2"
              title="Telakkakatu 1 A 1"
            >
              Telakkakatu 1 A 1
            </a>
          </span>
        </div>
        <div
          class="labelValuePair"
        >
          <span
            class="label standard"
          >
            Pituus:
          </span>
          <span
            class="value left"
          >
            6,0 m
          </span>
        </div>
        <div
          class="labelValuePair"
        >
          <span
            class="label standard"
          >
            Leveys:
          </span>
          <span
            class="value left"
          >
            3,0 m
          </span>
        </div>
        <div
          class="labelValuePair"
        >
          <span
            class="label standard"
          >
            Syvyys:
          </span>
          <span
            class="value left"
          >
            -
          </span>
        </div>
        <div
          class="labelValuePair"
        >
          <span
            class="label standard"
          >
            Kiinnitystapa:
          </span>
          <span
            class="value left"
          >
            Kävelyaisapaikka
          </span>
        </div>
        <div
          class="labelValuePair"
        >
          <span
            class="label standard"
          >
            Voimassa:
          </span>
          <span
            class="value left"
          >
            20.05.2019 - 15.06.2020
          </span>
        </div>
      </section>
    </article>
    <div>
      Mocked
    </div>
  </div>
  <div
    class="cardBody createLease"
  >
    <button>
      <span
        class="text brand span"
      >
        Lisää uusi
      </span>
    </button>
  </div>
   
</div>
`;
